### PR TITLE
sysdeps/linux: implement syscall macro

### LINKED ISSUE #523: Implement the `syscall` macro on Linux
Given that the alternative is patching everything, this is probably the best option. We might be able to rip this from musl, or maybe it'd be not too hard to implement ourselves.

### PR DESCRIPTION
**Pull request by 64**
Fixes #523.

### CODE REVIEW

**Comment by 64** [comment]
Hmm... the I guess that `__linux__` is still defined when compiling the other sysdeps on CI, so we can't use a simple `#ifdef __linux__` guard to pull in the linux syscall macros. Any ideas? Maybe we should use a different macro that can be detected and set by meson.

**Comment by avdgrinten** [comment]
Can't we pass `-U __linux__` when compiling the other sysdeps?

File: tests/meson.build [hunk 1]
```diff
@@ -79,7 +79,8 @@ posix_fail_test_cases = [
 
 glibc_test_cases = [
 	'getopt',
-	'ffsl-ffsll'
+	'ffsl-ffsll',
+	'syscall',
 ]
```
Comment (avdgrinten) [inline tests/meson.build:84]: This test only works on Linux, it should probably be called `linux-syscall` and only be added if we are compiling for Linux.

File: options/posix/include/unistd.h [hunk 2]
```diff
@@ -12,6 +12,10 @@
 #include <abi-bits/gid_t.h>
 #include <abi-bits/pid_t.h>
 
+#ifdef __linux__
+#include <mlibc/syscall.h>
+#endif
```
Comment (avdgrinten) [inline options/posix/include/unistd.h:17]: `mlibc/` are internal headers that are not installed. This should be in `bits/.

Comment (avdgrinten) [reply options/posix/include/unistd.h:17]: And the header also needs to be installed.

**Review by avdgrinten — APPROVED** [review]

**Pull request merged**